include/exadg/fluid_structure_interaction_precice/driver_fluid.h

```diff
@@ -0,0 +1,593 @@
+          "ALE-Mesh",
+          "Displacement");
+      }
+      else
+      {
+        AssertThrow(false, ExcNotImplemented());
+      }
+    }
+
+    // structure to fluid
+    {
+      std::vector<unsigned int> quad_indices;
+      quad_indices.emplace_back(fluid_operator->get_quad_index_velocity_linear());
+      quad_indices.emplace_back(fluid_operator->get_quad_index_velocity_nonlinear());
+      quad_indices.emplace_back(fluid_operator->get_quad_index_velocity_gauss_lobatto());
+
+      VectorType velocity_structure;
+      fluid_operator->initialize_vector_velocity(velocity_structure);
+      communicator_fluid = std::make_shared<InterfaceCoupling<dim, dim, Number>>(this->precice);
+      communicator_fluid->setup(
+        fluid_matrix_free,
+        fluid_operator->get_dof_index_velocity(),
+        quad_indices,
+        this->application->get_boundary_descriptor_fluid()->velocity->dirichlet_mortar_bc,
+        "Fluid-Mesh-read",
+        "Velocity");
+      // TODO: Change to Velocity
+
+      VectorType initial_stress;
+      fluid_operator->initialize_vector_velocity(initial_stress);
+      initial_stress = 0;
+      this->precice->initialize_precice(initial_stress);
+    }
+
+
+    /*********************************** INTERFACE COUPLING *************************************/
+
+
+    /**************************** SETUP TIME INTEGRATORS AND SOLVERS ****************************/
+
+    // fluid
+    {
+      // setup time integrator before calling setup_solvers (this is necessary since the setup
+      // of the solvers depends on quantities such as the time_step_size or gamma0!!!)
+      AssertThrow(this->application->get_parameters_fluid().solver_type ==
+                    IncNS::SolverType::Unsteady,
+                  ExcMessage("Invalid parameter in context of fluid-structure interaction."));
+
+      // initialize fluid_time_integrator
+      fluid_time_integrator =
+        IncNS::create_time_integrator<dim, Number>(fluid_operator,
+                                                   this->application->get_parameters_fluid(),
+                                                   this->mpi_comm,
+                                                   this->is_test,
+                                                   fluid_postprocessor);
+
+      fluid_time_integrator->setup(this->application->get_parameters_fluid().restarted_simulation);
+
+      fluid_operator->setup_solvers(
+        fluid_time_integrator->get_scaling_factor_time_derivative_term(),
+        fluid_time_integrator->get_velocity());
+    }
+
+    /**************************** SETUP TIME INTEGRATORS AND SOLVERS ****************************/
+  }
+
+  void
+  solve() const final
+  {
+    Assert(this->application->get_parameters_fluid().adaptive_time_stepping == false,
+           ExcNotImplemented());
+
+    bool is_new_time_window = true;
+
+    dealii::Timer precice_timer;
```

Comment:
why do you define the timer here? should make no difference when defining it in line 398, but would make the code more readable. The line `timer.restart()` should then not be necessary, already done by the constructor as far as I know.

Same for `DriverSolid`